pkg/orchestrator/gitHubActions.go

```diff
@@ -27,8 +74,47 @@ func (g *GitHubActionsConfigProvider) GetBuildStatus() string {
 }
 
 func (g *GitHubActionsConfigProvider) GetLog() ([]byte, error) {
-	log.Entry().Infof("GetLog() for GitHub Actions not yet implemented.")
-	return []byte{}, nil
+	ids, err := g.GetStageIds()
+	if err != nil {
+		return nil, err
+	}
+
+	logs := Logs{
+		b: make([][]byte, len(ids)),
+	}
+
+	ctx := context.Background()
+	sem := semaphore.NewWeighted(10)
+	wg := errgroup.Group{}
+	for i := range ids {
+		i := i // https://golang.org/doc/faq#closures_and_goroutines
+		if err := sem.Acquire(ctx, 1); err != nil {
+			return nil, fmt.Errorf("failed to acquire semaphore: %w", err)
+		}
+		wg.Go(func() error {
+			defer sem.Release(1)
+			resp, err := g.client.GetRequest(fmt.Sprintf("%s/jobs/%d/logs", getActionsURL(), ids[i]), g.getHeader(), nil)
+			if err != nil {
+				return fmt.Errorf("failed to get API data: %w", err)
+			}
+
+			b, err := io.ReadAll(resp.Body)
+			if err != nil {
+				return fmt.Errorf("failed to read response body: %w", err)
+			}
+			defer resp.Body.Close()
+			logs.Lock()
+			defer logs.Unlock()
+			logs.b[i] = append([]byte{}, b...)
```

Comment:
Shouldn't this be the same as:

```suggestion
			logs.b[i] = b
```